app/api/v1/users.rb

```diff
@@ -0,0 +1,19 @@
+class V1::Users < Grape::API
+  
+  before do
+    authenticate!
+    
+    @allowed_users = current_team&.users || User.all
```

Comment:
Nope, this is a vulnerability. In case there is no team selected, it returns everything. If there is a bug and there is no team, the api must return zero data back.